core/src/main/java/org/apache/metamodel/data/ScalarFunctionRow.java

```diff
@@ -49,8 +51,32 @@ public Object getValue(int index) throws IndexOutOfBoundsException {
             return _row.getValue(index - scalarFunctionCount);
         }
         final SelectItem selectItem = scalarFunctionSelectItems.get(index);
-        final SelectItem selectItemWithoutFunction = selectItem.replaceFunction(null);
-        return selectItem.getScalarFunction().evaluate(_row, selectItem.getFunctionParameters(), selectItemWithoutFunction);
+        if (selectItem.getScalarFunction().equals(FunctionType.CONCAT)) {
```

Comment:
Another "if concat" ... I would really prefer to get rid of these since a function should be pluggable and not require code changes in the "framework" if you will.